### DISCUSSION TITLE
How to Copy a Feature Service between Organizations (not just a reference)

### DISCUSSION THREAD

**Discussion by JoeDBryant** [EsriPS/arcgis-assistant-feedback#63] [Show and tell]
It would be helpful to have a step by step guide for copying feature services between organizations, much like you already have for copying Story Maps in your Guide Docs. This guide should show you how to change the feature service data from being referenced back to the original location.

I haven't tried this workflow yet, but I imagine it will require downloading the feature service(s) as a file geodatabase and republishing that data to the new organization as a hosted feature service from ArcGIS Pro. Then one could copy the feature layer(s) between organizations with the existing ArcGIS Assistant tool, and update the JSON of the feature layer with the URL of the newly published service. Having not done this before, I'm sure I will find something that doesn't translate, like the symbology, pop-ups, labels, or domain values. A guide written by someone who has done this before would be great to save all of us time and frustration.

The end goal would be to duplicate an app from the ground up by copying the feature service data, then layers, then the maps, then the app itself, each time updating the JSON with the new URLs.

**Comment by jpeterson** [comment]
Hey @JoeDBryant, thanks for starting this thread. I agree a guide for this workflow could be really useful at this point in time. That said, [our intention is to add feature service cloning to Assistant](https://guide.assistant.esri-ps.com/docs/working-with-items#services-feature-service-map-service-image-service-etc), which would eliminate the need for all the manual work you describe.

Also, for what it's worth – Assistant's predecessor, [ago-assistant](https://guide.assistant.esri-ps.com/docs/working-with-items#services-feature-service-map-service-image-service-etc), does have the capability to clone feature services.

**Comment by tsmcgrath** [comment]
@jpeterson I could use this guide today. It looks like copying feature services between orgs is still a copy by reference. Same ID's. I want to copy everything, data, feature service, map, app, story map from one org to another. And have it all work and use the storage credits from the target org - not the origination org. There's very little info on the specifics of that workflow.

**Comment by tsmcgrath** [comment]
@jpeterson Is this a fair high level summary?

Defining the remaining work to deploy

- Feature Services, Maps, Apps, and Story Maps are copied as clones, not copy-by-reference.
- Data is not cloned. Copy by reference only.
- All the data has to be loaded. New ID’s generated. Those new ID’s need to be used in cloned Feature Service, Map, App, and Story Map.
